addons/survey/wizard/survey_invite.py

```diff
@@ -165,6 +165,14 @@ def _compute_body(self):
                 invite = invite.with_context(lang=langs.pop())
             super(SurveyInvite, invite)._compute_body()
 
+    @api.depends('template_id')
+    def _compute_attachment_ids(self):
+        for invite in self:
+            if invite.template_id:
+                invite.attachment_ids = invite.template_id.attachment_ids
+            elif not invite.template_id.attachment_ids:
+                invite.attachment_ids = False
```

Comment:
The `elif`'s condition was always `True` because there couldn't be a `template_id` anymore.

If what you want to do is only changing when a template is added (avoid removing the attachments if the template is removed without being replaced), you should simply do this because `False` will be set above as well.
```suggestion
```